### ISSUE TITLE
bug(webpack): Typescript emitted no output

### ISSUE THREAD

**Issue by CaerusKaru** [angular/preboot#41]
* **I'm submitting a ...**
- [x] bug report
- [ ] feature request

* **Which parts of preboot are affected by this issue?**
- [ ] server side
- [ ] client side
- [ ] inline
- [x] build process
- [ ] docs
- [ ] tests

* **Do you want to request a *feature* or report a *bug*?**

Bug report

* **What is the current behavior?**

* **If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem** by creating a github repo.

1. Prepare a Universal bundle as specified [here](https://github.com/MarkPieszak/angular-cli/blob/70d814a2f439d26d6d6ffbae55191e49353b6b85/docs/documentation/stories/universal-rendering.md#angular-universal-integration)
2. Attempt to compile using `npm run build:dynamic`
3. Receive the below error

* **What is the expected behavior?**

Should be able to compile a bundle that relies on preboot

* **What is the motivation / use case for changing the behavior?**

Unable to use webpack to build a nodejs bundle from a TypeScript source

* **Please tell us about your environment:**

- Language: [**all** | TypeScript 2.5.2 | ES6/7 | ES5 ]
- OS:  [all | Mac OS X | Windows | **Linux** ]
- Platform: [all | **NodeJs** | Java | PHP | .Net | Ruby]

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, gitter, etc)

```
ERROR in ./node_modules/preboot/src/server/server-preboot.module.ts
Module build failed: Error: Typescript emitted no output for /home/debian/angular2/ngvirtualgrade/node_modules/preboot/src/server/server-preboot.module.ts.
You should not need to recompile .ts files in node_modules.
Please contact the package author to advise them to use --declaration --outDir.
More https://github.com/Microsoft/TypeScript/issues/12358
```

**Comment by jeffwhelpley** [comment]
I have not seen this one before. I will have time tomorrow to try this out and I will let you know.

**Comment by iamonuwa** [comment]
@jeffwhelpley this error still persists

**Comment by brexis** [comment]
Any idea on how to solve this?

**Comment by jeffwhelpley** [comment]
Apologies. This is a solvable issue. I just haven't had time to dive in. Since there are now at least 3 people running into this, I will bump up the priority on my list and get to it later this week.

**Comment by brexis** [comment]
@jeffwhelpley thank you

**Comment by jeffwhelpley** [comment]
hey, are you guys using ts-loader? what version of TypeScript? I started working on this today and will be publishing a new version of Preboot later tonight, but I could not recreate the error. It may be because this only affects certain versions of TypeScript:

https://github.com/TypeStrong/ts-loader/issues/278

**Comment by CaerusKaru** [comment]
Versions:
```
├── ts-loader@2.3.7 
└── typescript@2.5.3 
```

**Comment by jeffwhelpley** [comment]
I am pushing out another version shortly, but I noticed that a fix to this is to very simply change this line in your webpack.server.config.js:

```
resolve: { extensions: [".ts", ".js"] },
```

to this:

```
resolve: { extensions: [".js", ".ts"] },
```

I am still testing now, though, and will post any additional fixes.

**Comment by jeffwhelpley** [comment]
OK, I just published new version `5.1.4` but I believe you do need to make the fix I mention above as well.

**Comment by JoseFMP** [comment]
For me this issue is still happening.
